### PR TITLE
Allows maintenance jack to be quick drawn

### PR DESCRIPTION
**Pull request by Git-Nivrak**
# About the pull request
Allows maintenance jack to be quick drawn as well as adding a flag that allows to replicate this behavior for other items.

# Explain why it's good for the game
Quality of life, Requested by a player
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
qol: Maintenance jack can now be quick drawn
code: Added a flag that allows items to be quick draw-able despite not being weapons.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun_helpers.dm [hunk 1]
```diff
@@ -518,10 +518,10 @@ DEFINES in setup.dm, referenced here.
 
 	if(istype(slot) && (slot.storage_flags & STORAGE_ALLOW_QUICKDRAW))
 		for(var/obj/cycled_weapon in slot.return_inv())
-			if(isweapon(cycled_weapon))
+			if(isweapon(cycled_weapon) || cycled_weapon.flags_atom & QUICK_DRAWABLE)
```
Comment (Zonespace27) [inline code/modules/projectiles/gun_helpers.dm:521]: Honestly just remove the weapon check and slap the quickdraw flag on all weapons

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_helpers.dm (hunk 1 above)
Comment (Git-Nivrak) [reply code/modules/projectiles/gun_helpers.dm:521]: yea just thought of that

File: code/game/objects/items/tools/maintenance_tools.dm [hunk 2]
```diff
@@ -511,6 +511,7 @@
 
 /obj/item/maintenance_jack/Initialize()
 	. = ..()
+	flags_atom |= QUICK_DRAWABLE
```
Comment (Zonespace27) [inline code/game/objects/items/tools/maintenance_tools.dm:514]: why not just put this in the definition itself? Same can be argued for doing the same for `/obj/item/weapon` but i'm slightly less pushy on that one

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/tools/maintenance_tools.dm (hunk 2 above)
Comment (Git-Nivrak) [reply code/game/objects/items/tools/maintenance_tools.dm:514]: To not override other flags by accident

**Review by Drulikar — CHANGES_REQUESTED** [review]
All overrides to flags_atom need to be looked at.

For example `/obj/item/weapon/gun/launcher/rocket` can normally go into your suit slot and you can press E to draw it, but since `/obj/item/weapon/gun` overrides flags_atom it now no longer does so since you need the `QUICK_DRAWABLE` flag, and it doesn't have it.

You may consider using [`::`](https://www.byond.com/docs/ref/#/operator/::) to |= existing flags in definitions; but honestly there aren't that many to fix. The regex to find all weapon definitions in this scenario is probably something like `^/obj/item/weapon/[\w/]*?$((.|\n)(?!^/))*?flags_atom ?=(?!^/)` 
image

**Review by Drulikar — APPROVED** [review]

**Pull request merged**